### DISCUSSION TITLE
Apply shear traction

### DISCUSSION THREAD

**Discussion by ndaman** [idaholab/moose#20647] [Q&A Modules: Solid mechanics] [answered]
I feel like I am misunderstanding the pressure BC system, I am trying to apply shear tractions on outer boundary of a 2D square (to compare with the homogenization system with periodic boundaries). The relevant BC's that I have in my input file are:

   [top]
      type = Pressure
      boundary = top
      factor = 1
      variable = disp_x
      component = 0
    []
    [right]
      type = Pressure
      boundary = right
      factor = 1
      variable = disp_y
      component = 1
    []
    [bottom]
      type = Pressure
      boundary = bottom
      factor = -1
      variable = disp_x
      component = 0
    []
    [left]
      type = Pressure
      boundary = left
      factor = -1
      variable = disp_y
      component = 1
  []

When run this way, the model appears to have no tractions at all. If we focus on just the top (with a normal vector in the positive y direction), how would I apply a traction that acts in the positive x-direction only?

**Comment by hugary1995** [comment, marked as answer]
You want to use NeumannBCs

**Reply by ndaman** [reply]
Ah, thanks!
